crates/sc-subspace-block-relay/src/consensus/types.rs

```diff
@@ -103,3 +106,97 @@ impl<Block: BlockT> PartialBlock<Block> {
         }
     }
 }
+
+/// Client side metrics.
+pub(crate) struct ConsensusClientMetrics {
+    pub(crate) requests: RelayCounter,
+    pub(crate) failed_requests: RelayCounterVec,
+    pub(crate) blocks_downloaded: RelayCounter,
+    pub(crate) bytes_downloaded: RelayCounter,
+    pub(crate) tx_pool_miss: RelayCounter,
+}
+
+impl ConsensusClientMetrics {
+    pub(crate) fn new(registry: Option<&Registry>) -> Result<Self, PrometheusError> {
+        Ok(Self {
+            requests: RelayCounter::new(
+                "relay_client_requests",
+                "Total number of requests initiated by client",
+                registry,
+            )?,
+            failed_requests: RelayCounterVec::new(
+                "relay_client_failed_requests",
+                "Number of failed client requests",
+                &[ERR_LABEL],
+                registry,
+            )?,
```

Comment:
Usually when you query these metrics and build dashboard, it is useful to have metrics that are called similarly, so here second should probably be called `requests_failed` for them to appear next to each other in outocomplete.

However, I think this should be one metric in general with a label like `result` that is either `success` or one of error variants. That way you can both know total number of requests by just looking at sum of all `requests` as well as split them by labels to see breakdown into all variants and having both success and all minus success is very convenient.

Reply:
Makes sense, consolidated the counters under CounterVec everywhere